### PR TITLE
Blackjaxified Fig 15.8 `logreg_iris_bayes_2d`

### PR DESCRIPTION
**Pull request by karm-patel**
## Description
Blackjaxified `logreg_iris_bayes_2d`

### Figure Number
Fig 15.8

### Figures
Before:
image

After:
image

### CODE REVIEW

File: notebooks/book2/15/logreg_iris_bayes_2d.ipynb [hunk 1]
```diff
@@ -1,49 +1,229 @@
 {
```
Comment (murphyk) [inline notebooks/book2/15/logreg_iris_bayes_2d.ipynb:1]: <p>I think we should keep a comment pointing to the original pymc code at</p><p> </p><p>https://github.com/aloctavodia/BAP/blob/master/code/Chp4/04_Generalizing_linear_models.ipynb</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/probml/pyprobml/pull/994/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (murphyk) [inline notebooks/book2/15/logreg_iris_bayes_2d.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #17.        return logprob_prior + logprob_likeli
</pre><p>I would rename to <code>log_prior</code> and <code>log_lik</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/probml/pyprobml/pull/994/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (murphyk) [inline notebooks/book2/15/logreg_iris_bayes_2d.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #9.    jax.vmap(joint_logprob_fn_partial)(initial_params)
</pre><p>This all seems much more complex than the pymc code. I wonder if BIJAX could be used to simplify this boilerplate?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/probml/pyprobml/pull/994/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (murphyk) [inline notebooks/book2/15/logreg_iris_bayes_2d.ipynb:1]: <p>what is <code>x1_all</code>?Is it the same as <code>bd</code>,the decision boundary?    </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/probml/pyprobml/pull/994/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (karm-patel) [reply notebooks/book2/15/logreg_iris_bayes_2d.ipynb:1]: <p>Done</p>

Comment (karm-patel) [reply notebooks/book2/15/logreg_iris_bayes_2d.ipynb:1]: <p>Ohh I missed it, Done.</p>

Comment (karm-patel) [reply notebooks/book2/15/logreg_iris_bayes_2d.ipynb:1]: <p>Ohh, actually we don't need this line, this was just for sanity check. I've removed it.</p><p> </p><p>I think this is too simple example for BIJAX,  would it make more sense to show usage of BIJAX in little more complex example like hierarchical one where we needed bijectors?  I'll add BIJAX implementation in those examples.</p><p> </p>

Comment (karm-patel) [reply notebooks/book2/15/logreg_iris_bayes_2d.ipynb:1]: I've renamed it to `bd` and added a comment to avoid confusion.

**Pull request merged**